### PR TITLE
Made some changes to the Packer templating docs

### PR DESCRIPTION
**Pull request by jamtur01**
1. Added the `user` function.
2. Some tidying up of language.

### CODE REVIEW

**Comment by jamtur01** [comment]
Also it's not clear to me what `{{.Variables}}` refer to if variables are specified via `{{user \`variable\`}}`? Can someone clarify?

**Comment by mwhooker** [comment]
@jamtur01 I believe `{{.Variables}}` are specific variables defined in the template processor that are automatically available in some contexts. For example, in the [shell provisioner, there are `{{.Vars}}` and `{{.Path}}` variables available](https://www.packer.io/docs/provisioners/shell.html#sudo-example).

Perhaps variable is a misleading word, since they're not "variable" in the normal sense

Thanks for the patch!

File: website/source/docs/templates/engine.html.md [hunk 1]
```diff
@@ -18,13 +18,12 @@ For example, the `{{timestamp}}` function can be used in any string to generate
 the current timestamp. This is useful for configurations that require unique
 keys, such as AMI names. By setting the AMI name to something like
 `My Packer AMI {{timestamp}}`, the AMI name will be unique down to the second.
-If you need greater than one second granularity, you should use `{{uuid}}`, as
-when you have multiple builders in the same template.
+If you need greater than one second granularity, you should use `{{uuid}}`, for example when you have multiple builders in the same template.
```
Comment (mwhooker) [inline website/source/docs/templates/engine.html.md]: Could you line break here?

**Comment by jamtur01** [comment]
Ah okay! That makes sense. Maybe a reword? Seems like that's a parallel use of `variable` in a confusing sense?

**Comment by mwhooker** [comment]
I think a reword might be a good idea. I'm just thinking back to every document I've read about templating engines, and they all refer to the same concept of variables. Do you have any ideas? Maybe "template values". `{{.Value}}`...

**Comment by jamtur01** [comment]
Yeah was just trying out a few words:

`user variables`
`template values`
`builder variables`
`component variables`

?

**Comment by mwhooker** [comment]
I'm in favor of "template variables", which I think distinguishes them from "user variables". On second thought I think it would be incorrect to call them anything but variables, because they do change from run to run

**Comment by jamtur01** [comment]
Okay. I'll do a pass and update the language if that's cool?

**Comment by mwhooker** [comment]
much appreciated =)

**Pull request merged**